### PR TITLE
E2e test test af search result app

### PR DESCRIPTION
**Pull request by spaceo**
#### Link to issue

https://reload.atlassian.net/browse/DDFSOEG-189

#### Description

Adding cypress tests for search result

- Introducing usage of local images in fixtures. Will also increase speed.
- Adding fixture data for a cover
- Adding fixture data for the search result test itself

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: src/apps/search-result/search-result.test.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    cy.log("Check length of search result list:");
+    cy.get(".search-result-page__list").find("li").should("have.length", 2);
+
+    cy.log("Do the search results have images?");
+    cy.get(".search-result-page__list .search-result-item img")
+      .should("have.attr", "src")
+      .and("match", coverUrlPattern);
+
+    cy.log("Does the search result have favourite buttons?");
```
Comment (kasperg) [inline src/apps/search-result/search-result.test.ts:17]: I like this way of attaching something humanly readable to an assertion. It reminds me a bit of PHPUnit 😄 . We could also do this with code comments but this puts it in the Cypress UI which may make it easier to work with in GitHub Actions in the like.

File: src/apps/search-result/search-result.test.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    cy.get(
+      ".search-result-page__list .search-result-item .button-favourite"
+    ).should("have.attr", "aria-label", "Add to favorites");
+
+    cy.log("Does the search result have a series line?");
+    cy.get(
+      ".search-result-page__list .search-result-item .horizontal-term-line"
+    ).should(
+      "contain.text",
+      "Nr. 1  i serienDummy Some SeriesNr. 1  i serienDummy Some Series"
```
Comment (kasperg) [inline src/apps/search-result/search-result.test.ts:27]: Consider adding a comment explaining why this text is somewhat odd.

An alternative approach would be to modify the mocked response to return something that is closer to what we expected.

File: src/apps/search-result/search-result.test.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    cy.log("Does the search result have titles?");
+    cy.get(".search-result-page__list .search-result-item h2").should(
+      "contain.text",
+      "Dummy Some Title: Full"
+    );
+
+    cy.log("Does the search result have authors?");
+    cy.get(
+      ".search-result-page__list .search-result-item p.text-small-caption"
+    ).should("contain.text", "Af Dummy Jens Jensen (Dummy 1839)");
```
Comment (kasperg) [inline src/apps/search-result/search-result.test.ts:39]: Consider reducing the dependency on CSS selectors.

Cypress and others recommend testing based on what is visible to end users or assistive technologies to reduce frailty. In this regard it is sufficient that the UI contains the text "Af Dummy Jens Jensen (Dummy 1839)".

This also applies for other tests in this PR.

**Review by kasperg — COMMENTED** [review]
:+1: Good stuff.

File: src/apps/search-result/search-result.test.ts (hunk 1 above)
Comment (spaceo) [reply src/apps/search-result/search-result.test.ts:17]: 👍

File: src/apps/search-result/search-result.test.ts (hunk 2 above)
Comment (spaceo) [reply src/apps/search-result/search-result.test.ts:27]: Yes UI agree. I think we need to revise the text when we get some meaningful data from DBC. I'll write a comment about that.

File: src/apps/search-result/search-result.test.ts (hunk 3 above)
Comment (spaceo) [reply src/apps/search-result/search-result.test.ts:39]: Check

**Review by cableman — APPROVED** [review]

**Pull request merged**